src/client/renderer/r_light.c

```diff
@@ -182,29 +185,19 @@ void R_EnableLights(uint64_t mask) {
 	if (mask) { // enable up to MAX_ACTIVE_LIGHT sources
 		const r_light_t *l = r_view.lights;
 
-		vec4_t position = { 0.0, 0.0, 0.0, 1.0 };
-		vec4_t diffuse = { 0.0, 0.0, 0.0, 1.0 };
-
 		for (uint16_t i = 0; i < r_view.num_lights; i++, l++) {
 
-			if (j == MAX_ACTIVE_LIGHTS)
+			if (j == r_state.max_lights)
```

Comment:
Not sure about that one. You should never call UseLight with a value that is beyond r_state.max_lights, it will crash. The game is set up to not push more than the max hardware lights that is allowed by the program.